### PR TITLE
Feature/add endpoint to check if ssn has signed product already

### PR DESCRIPTION
**Pull request by AlexSJ1990**

### CODE REVIEW

File: member-service/src/main/java/com/hedvig/memberservice/web/dto/SsnAlreadySignedMemberResponse.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.hedvig.memberservice.web.dto
+
+data class SsnAlreadySignedMemberResponse (
+    val ssnAlreadyMember: Boolean
```
Comment (meletios) [inline member-service/src/main/java/com/hedvig/memberservice/web/dto/SsnAlreadySignedMemberResponse.kt]:
```suggestion
    val isSsnAlreadySignedMember: Boolean
```

File: member-service/src/main/java/com/hedvig/memberservice/web/v2/SignController.java [hunk 2]
```diff
@@ -47,6 +48,13 @@
     }
   }
 
+  @GetMapping("signedSSN")
+  public ResponseEntity<SsnAlreadySignedMemberResponse> ssnAlreadySigned(@RequestHeader String ssn) {
+
+    val alreadySigned = signingService.ssnAlreadySignedMember(ssn);
```
Comment (meletios) [inline member-service/src/main/java/com/hedvig/memberservice/web/v2/SignController.java]:
```suggestion
    val isSsnAlreadySignedMember = signingService.ssnAlreadySignedMember(ssn);
```

File: member-service/src/main/java/com/hedvig/memberservice/web/v2/SignController.java [hunk 3]
```diff
@@ -47,6 +48,13 @@
...
+    return ResponseEntity.ok(alreadySigned);
```
Comment (meletios) [inline member-service/src/main/java/com/hedvig/memberservice/web/v2/SignController.java]:
```suggestion
    return ResponseEntity.ok(isSsnAlreadySignedMember);
```

File: member-service/src/main/java/com/hedvig/memberservice/web/dto/SsnAlreadySignedMemberResponse.kt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package com.hedvig.memberservice.web.dto
+
+data class SsnAlreadySignedMemberResponse (
```
Comment (meletios) [inline member-service/src/main/java/com/hedvig/memberservice/web/dto/SsnAlreadySignedMemberResponse.kt]:
```suggestion
data class IsSsnAlreadySignedMemberResponse (
```

File: member-service/src/main/java/com/hedvig/memberservice/services/SigningService.java [hunk 5]
```diff
@@ -147,6 +148,14 @@ public MemberSignUnderwriterQuoteResponse signUnderwriterQuote(final long member
     }
   }
 
+  public SsnAlreadySignedMemberResponse ssnAlreadySignedMember(final String ssn) {
+
+    Optional<SignedMemberEntity> existing = signedMemberRepository.findBySsn(ssn);
+    return existing.isPresent()
+      ? new SsnAlreadySignedMemberResponse(true)
+      : new SsnAlreadySignedMemberResponse(false);
+  }
```
Comment (palmenhq) [inline member-service/src/main/java/com/hedvig/memberservice/services/SigningService.java]: Maybe 
```java
return new SsnAlreadySignedMemberResponse(exisiting.isPresent())
```

File: member-service/src/main/java/com/hedvig/memberservice/services/SigningService.java [hunk 6]
```diff
@@ -147,6 +148,14 @@ public MemberSignUnderwriterQuoteResponse signUnderwriterQuote(final long member
     }
   }
 
+  public SsnAlreadySignedMemberResponse ssnAlreadySignedMember(final String ssn) {
```
Comment (palmenhq) [inline member-service/src/main/java/com/hedvig/memberservice/services/SigningService.java]: I think we could improve this name by prefixing it with `is` :) in that way the method becomes a verb, and its clear what it'll do

**Pull request merged**